### PR TITLE
Dashboard date filter

### PR DESCRIPTION
**Pull request by EDsCODE**
*Implemented only date filtering first

### CODE REVIEW

**Comment by EDsCODE** [comment]
[Dashboard Filtering](https://trello.com/c/B3xMpXbE/63-dashboard-filtering)

**Comment by EDsCODE** [comment]
TODOs left:
- style filters so it's prettier and more intuitive
- decide on how to filter credit transfers by account type

Notes: I was able to pipe the requested filters from the front end all the way to calculate_transfer_stats in credit_transfer.py. However, i'm not sure how to proceed with filtering directly on the backend because attributes such as "balance" are computed properties so I can't use db.query filters directly on transferaccount balances.

File: app/server/utils/credit_transfer.py [hunk 1]
```diff
@@ -36,56 +36,75 @@ def dollars_to_cents(amount_dollars):
...
+        dateFilter.append(CreditTransfer.created <= end_date)
+
+    filters = [CreditTransfer.transfer_type == TransferTypeEnum.PAYMENT, CreditTransfer.transfer_subtype == TransferSubTypeEnum.DISBURSEMENT]
+    
     total_distributed = (
         db.session.query(func.sum(CreditTransfer.transfer_amount).label('total'))
-            .filter(CreditTransfer.transfer_type == TransferTypeEnum.PAYMENT)
-            .filter(CreditTransfer.transfer_subtype == TransferSubTypeEnum.DISBURSEMENT).first().total
-    )
+            .filter(*filters).filter(*dateFilter).first().total
```
Comment (enjeyw) [inline app/server/utils/credit_transfer.py]: nice

**Comment by enjeyw** [comment]
The charcoal on tealish is pretty hard to read (though i'm admittedly blind)
image

I'd probably advocate for either the standard white background dropdowns (think this is probably best), or using white text for a tealish background (i also matched the colour to our logo here (it's sliiggghhhtttly brighter), because the current colour is from our older style)

image

**Comment by enjeyw** [comment]
if it's not a massive pain, might be good to get the date picker and the filter module inlined? feels like a lot of wasted space to the right of the picker currently...

**Comment by enjeyw** [comment]
> TODOs left:
> 
> * style filters so it's prettier and more intuitive
> * decide on how to filter credit transfers by account type
> 

ohhh shucks sorry forgot u wrote these TODOs - my comments are probably redundant then; soz.

File: app/server/api/credit_transfer_api.py [hunk 2]
```diff
@@ -449,6 +450,28 @@ def post(self):
...
 
+class CreditTransferStatsApi(MethodView):
+    @requires_auth(allowed_roles={'ADMIN': 'any'})
+    def get(self):
+
+        start_date = request.args.get('start_date')
+        end_date = request.args.get('end_date')
+        encoded_filters = request.args.get('filters')
+        encoded_filters = encoded_filters.replace(" ", "+")
+        decoded_filters = base64.b64decode(encoded_filters)
```
Comment (enjeyw) [inline app/server/api/credit_transfer_api.py]: this is probably something we worry about later, but my gut is that we shouldn't need to encode the filters to get them to play nice - query strings are pretty good at handling this sort of thing, and more cannonical IMO. Eg like https://www.carsales.com.au/retailapi/home/v2/carsales/search?makePopular=Mazda&model=3&stateList=South%20Australia&price=%3B60000&&_a=all&_t=price

**Comment by EDsCODE** [comment]
> > TODOs left:
> > 
> > * style filters so it's prettier and more intuitive
> > * decide on how to filter credit transfers by account type
> 
> ohhh shucks forgot u wrote these TODOs - my comments are probably redundant then; soz.

haha i'm definitely not requesting to push with styling like that. i'm currently restyling the date range picker and will inline it with the rest of the filters

File: app/server/api/credit_transfer_api.py (hunk 2 above)
Comment (EDsCODE) [reply app/server/api/credit_transfer_api.py]: yeah this is definitely unneeded. i had it temporarily to test the parameters serverside where i found the other blocker relating to computed properties but will change the params to separate key-values

**Comment by EDsCODE** [comment]
@enjeyw give this a look! it's just date filter for now, moderately styled. going to go setup a dev script that can mock a ton of data for testing filters/search/metrics

**Comment by EDsCODE** [comment]
[Dashboard-Date filtering](https://trello.com/c/tv31FDuO/323-dashboard-date-filtering)

**Comment by enjeyw** [comment]
Oh cool, will check out today!

Technically there’s a script under migrations called “dev_data.py” that creates a bunch of seed test data, but it’s TOTALLY broken lol. So you could either work off that, or start from scratch.

**Comment by EDsCODE** [comment]
[Dashboard Filtering](https://trello.com/c/B3xMpXbE/63-dashboard-filtering)

**Pull request merged**